networkx/algorithms/tests/test_planarity.py

```diff
@@ -372,71 +372,102 @@ def check_counterexample(G, sub_graph):
 
 
 class TestPlanarEmbeddingClass:
+    def test_add_half_edge(self):
+        embedding = nx.PlanarEmbedding()
+        embedding.add_half_edge(0, 1)
+        with pytest.raises(
+            nx.NetworkXException, match="Invalid clockwise reference node."
+        ):
+            embedding.add_half_edge(0, 2, cw=3)
+        with pytest.raises(
+            nx.NetworkXException, match="Invalid counterclockwise reference node."
+        ):
+            embedding.add_half_edge(0, 2, ccw=3)
+        with pytest.raises(
+            nx.NetworkXException, match="Only one of cw/ccw can be specified."
+        ):
+            embedding.add_half_edge(0, 2, cw=1, ccw=1)
+        with pytest.raises(
+            nx.NetworkXException,
+            match=(
+                r"Node already has out-half-edge\(s\), either"
+                " cw or ccw reference node required."
+            ),
+        ):
+            embedding.add_half_edge(0, 2)
+        # these should work
+        embedding.add_half_edge(0, 2, cw=1)
+        embedding.add_half_edge(0, 3, ccw=1)
```

Comment:
This smoke test could be beefed up by explicitly verifying the results, something like:

```python
assert sorted(embedding.edges(data=True)) == [
    (0, 1, {'ccw': 2, 'cw': 3}),
    (0, 2, {'cw': 1, 'ccw': 3}),
    (0, 3, {'cw': 2, 'ccw': 1}),
]

Reply:
Makes sense, I've added this check.